### PR TITLE
Merge pivot options

### PR DESCRIPTION
**Pull request by fdncred**
As per @thegedge suggestion on Discord, I merged the auto pivot options into an enum.
```rust
    enum AutoPivotMode {
        Auto,
        Always,
        Never,
        ToFit,
    }
```

### CODE REVIEW

File: crates/nu-cli/src/commands/autoview.rs [hunk 1]
```diff
@@ -234,8 +247,10 @@ pub fn autoview(context: RunnableContext) -> Result<OutputStream, ShellError> {
                                 yield Err(e);
                             }
 
-                            Value { value: UntaggedValue::Row(row), ..} if !no_auto_pivot
-                                || (pivot_to_fit && // Or if the row character count + number of headers * 2 (for padding) > terminal width
+                            Value { value: UntaggedValue::Row(row), ..} 
+                                if pivot_mode == AutoPivotMode::Always || 
+                                ((pivot_mode == AutoPivotMode::ToFit || 
+                                    pivot_mode == AutoPivotMode::Auto) &&
```
Comment (thegedge) [inline crates/nu-cli/src/commands/autoview.rs:253]: This makes me think `ToFit` and `Auto` are the same. What do you think?

Comment (fdncred) [reply crates/nu-cli/src/commands/autoview.rs:253]: I tried to implement it as you instructed.
```
I'm also thinking we should collapse the multiple config 
values into one: pivot_mode with values auto, always, 
never, to_fit (perhaps to_fit could be the auto)
```
So, `to_fit` is the `auto`.

**Comment by fdncred** [comment]
I also have no idea what that `fmt` error means. Take 3 lines out and replace them with the exact same 3 lines? I noticed that before I committed and figured someone would help me understand what to do here.

File: crates/nu-cli/src/commands/autoview.rs (hunk 1 above)
Comment (fdncred) [reply crates/nu-cli/src/commands/autoview.rs:253]: oh, perhaps you're saying to rename `to_fit` to `auto`. If that's what you mean, i can change it.

Comment (thegedge) [reply crates/nu-cli/src/commands/autoview.rs]: Yep, you got it 🙂

**Comment by thegedge** [comment]
@fdncred `rustfmt` really should highlight what rule is failing, which would simplify this, but looks like you have some trailing whitespace on one of the lines 🙂

**Comment by fdncred** [comment]
@thegedge Thanks. I would've never known that was a trailing space error.

**Comment by thegedge** [comment]
🙌 🚀 

This _will_ break things for anyone already using the `no_auto_pivot`, but I think it's a much clearer config value. Move fast and break things is kind of how nushell rolls at the moment, but we'll have to think about how we could transition these kinds of things in the future 🙂

**Pull request merged**